### DISCUSSION TITLE
[Feature] GPU Hardware acceleration for video transcoding

### DISCUSSION THREAD

**Discussion by auanasgheps** [immich-app/immich#2674] [Feature Request]
### The feature

I am following up on similar discussions that have been closed: #203 and maybe #307.

When importing a large gallery, I noticed that a lot of time is spent processing videos that have to be transcoded in the right format to be played by every device.

Would be wonderful to offload this task to the GPU, which is incredibly faster, even iGPUs. As an example, Plex and Jellyfin support this technology. 

Yes, GPU encoding is (usually) less quality accurate than CPU, but we're talking about 720p "playable previews".

AMD, NVIDIA and Intel all offer APIs for this scenario.

Another cool but advanced feature would be to encode these clips only when requested, in real time, exactly like Plex and Jellifin do.

I personally use this technology on Plex and Handbrake with a Intel integrated GPU and it makes a huge difference.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by alextran1502** [comment]
Hello, this is being worked on. We will announce the official support when the work is done. Thank you

**Reply by auanasgheps** [reply]
I really appreciate your dedication to this project and how fast you reply to users!

Immich will be my next OOS donation very soon :)

**Comment by auanasgheps** [comment]
It is implemented and I just tested it. Works great, thanks.

The procedure just needs to be "adjusted" for Portainer users
